src/main/java/at/hannibal2/skyhanni/features/inventory/caketracker/CakeTracker.kt

```diff
@@ -0,0 +1,424 @@
+        "cake.bag",
+        "New Year Cake Bag",
+    )
+
+    /**
+     * REGEX-TEST: Auctions Browser
+     * REGEX-TEST: Auctions: "Test"
+     * REGEX-TEST: Auctions: "New Year Cake (Year
+     */
+    private val auctionBrowserPattern by patternGroup.pattern(
+        "auction.search",
+        "Auctions Browser|Auctions: \".*",
+    )
+
+    /**
+     * REGEX-TEST: Auctions: "New Year C
+     */
+    private val auctionCakeSearchPattern by patternGroup.pattern(
+        "auction.cakesearch",
+        "Auctions: \"New Year C.*",
+    )
+
+    private fun addCake(cakeYear: Int) {
+        val storage = storage ?: return
+        if (cakeYear !in storage.ownedCakes) storage.ownedCakes.add(cakeYear)
+        recalculateMissingCakes()
+    }
+
+    private fun removeCake(cakeYear: Int) {
+        val storage = storage ?: return
+        if (cakeYear in storage.ownedCakes) storage.ownedCakes.remove(cakeYear)
+        recalculateMissingCakes()
+    }
+
+    private fun isEnabled() = LorenzUtils.inSkyBlock && config.enabled
+
+    @HandleEvent
+    fun onCommandRegistration(event: CommandRegistrationEvent) {
+        event.register("shresetcaketracker") {
+            description = "Resets the New Year Cake Tracker"
+            category = CommandCategory.USERS_RESET
+            callback {
+                val storage = storage ?: return@callback
+                storage.ownedCakes.clear()
+                recalculateMissingCakes()
+                ChatUtils.chat("New Year Cake tracker data reset")
+            }
+        }
+    }
+
+    @SubscribeEvent
+    fun onConfigLoad(event: ConfigLoadEvent) {
+        ConditionalUtils.onToggle(config.maxDisplayRows) {
+            lastKnownCakeDataHash = 0
+        }
+        ConditionalUtils.onToggle(config.unobtainedAuctionHighlightColor) {
+            unobtainedHighlightColor = config.unobtainedAuctionHighlightColor.get().toChromaColor()
+        }
+        ConditionalUtils.onToggle(config.obtainedAuctionHighlightColor) {
+            obtainedHighlightColor = config.obtainedAuctionHighlightColor.get().toChromaColor()
+        }
+    }
+
+    @SubscribeEvent
+    fun onBackgroundDraw(event: GuiRenderEvent.ChestGuiOverlayRenderEvent) {
+        if (!isEnabled()) return
+        if (inCakeBag || (inAuctionHouse && (auctionCakesCache.isNotEmpty() || searchingForCakes))) {
+            reRenderDisplay()
+        }
+    }
+
+    @SubscribeEvent
+    fun onBackgroundDraw(event: GuiContainerEvent.BackgroundDrawnEvent) {
+        if (!isEnabled()) return
+        if (inCakeInventory) checkInventoryCakes()
+
+        if (!inAuctionHouse) return
+        (event.gui.inventorySlots as ContainerChest).getAllItems().filter {
+            auctionCakesCache.containsKey(it.key.slotIndex) &&
+                cakeNamePattern.matches(it.key.stack.displayName)
+        }.forEach { (slot, _) ->
+            slot.getHighlightColor()?.let { slot highlight it }
+        }
```

Comment:
rather than looking at all items in the inventory every single frame, cache the slots and colors to get highlighted

also, the .getHighlightColor function already uses cakeNamePattern